session/pingpong/consumer_balance_tracker.go

```diff
@@ -49,8 +49,10 @@ func newBalanceKey(chainID int64, id identity.Identity) balanceKey {
 // ConsumerBalanceTracker keeps track of consumer balances.
 // TODO: this needs to take into account the saved state.
 type ConsumerBalanceTracker struct {
-	balancesLock sync.Mutex
-	balances     map[balanceKey]ConsumerBalance
+	balancesLock           sync.Mutex
+	balances               map[balanceKey]ConsumerBalance
+	transactorBountiesLock sync.Mutex
```

Comment:
Dont think you're supposed to do that.
* [gobyexample (mutexes)](https://gobyexample.com/mutexes)
* [tour of go](https://tour.golang.org/concurrency/9)

Point is: you already got a mutex, an additional one - doesn't really do anything.

Reply:
golang noob, checking it now